replay2/memory_remapper/memory_remapper.cc

```diff
@@ -14,4 +14,98 @@
 
 #include "memory_remapper.h"
 
-void foo(int var) { return; }
+namespace agi {
+namespace replay2 {
+
+	ReplayAddress MemoryRemapper::AddMapping(const MemoryObservation& observation)
+	{
+		const CaptureAddress captureAddress = observation.captureAddress();
```

Comment:
Yeah, I can make this a const reference. You're correct to say this causes a copy to the stack.

Reply:
Are there a way to detect this automatically? Maybe we can add a static analyser to our tests similar to Python. It can possibly catch all these things for free.

Reply:
We've talked about this offline and Melih will look into what tools we can use to catch stuff like this